app/Events/ThiefCaughtEvent.php

```diff
@@ -22,7 +22,10 @@ public function __construct(User $user)
     {
         $this->smokescreened_users = new Collection();
 
-        $this->user = $user;
+        $fullUser = $user->toArray();
+        $fullUser["role"] = $user->inviteKey->role;
+        $this->user = $fullUser;
+
```

Comment:
In dit event wordt ook een nieuw interval aangeroepen. Dit kan voor problemen zorgen, omdat spelers zichzelf nog moeten updaten voordat het interval plaatsvind. Om problemen te voorkomen en logica te scheiden stel ik voor om het `event(new GameIntervalEvent` uit dit event uit te halen.